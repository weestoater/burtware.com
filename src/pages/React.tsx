import React from "react";
import ReactHooksPost from "../components/react/ReactHooksPost";
import RandomUsersPost from "../components/react/RandomUserPost";

export default function ReactPage() {
  return (
    <div className="row">
      <RandomUsersPost />
      <ReactHooksPost />
      <div className="col-lg-3 col-md-6 mb-4">
        <div className="card">
          <div className="card-header">
            <i className="bi bi-filetype-jsx" />
            <span className="pl-4">my experiences</span>
          </div>
          <div className="card-body">
            <p>
              There are as many ways to do Single Page Applications (SPAs) as
              there are developers with opinions. Within my work we have many
              instances of AngularJS, Angular (4,5,6,7,9,11,12) and ReactJS.
            </p>
            <p>
              My experience with ReactJS began as pretty limited, but with a
              strong following wind, a few cups of coffee and a Pluralsight
              subscription â€“ I resolved that in no small measure.</p>
            <p>
              Before moving recently to pure UX team, I worked on a growing React platform which included multiple chart and grid offerings.
            </p>
            <p>
              I am now trying to get my head around the worlds of <code>typescript</code>; micro-services and AWS with React.
            </p>
          </div>
        </div>
      </div>
    </div>
  );
}
