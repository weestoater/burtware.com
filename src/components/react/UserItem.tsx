import React from "react";

function UserItem(props: any) {

  const user = props;

  return (
    <>
      <div className="user col-sm-12 mb-3">
        <img src={user.item.picture.large} alt="" className="picture" />
        <p className="name">
          <span className="badge bg-info text-black me-2">{user.item.nat}</span>{" "}
          {user.item.name.title} {user.item.name.first} {user.item.name.last}
        </p>
        <address className="address">
          {user.item.location.street.number} {user.item.location.street.name}{" "}
          <br />
          {user.item.location.city}
          <br />
          {user.item.location.state}, {user.item.location.country}
          <br />
          {user.item.location.postcode}
        </address>
        <p className="contact">
          <i className="bi bi-phone" aria-hidden="true" />{" "}
          <a href={`tel:${user.item.cell}`}>{user.item.cell}</a>
          <br />
          <i className="bi bi-telephone" aria-hidden="true" />{" "}
          <a href={`tel:${user.item.phone}`}>{user.item.phone}</a>
          <br />
          <i className="bi bi-envelope" aria-hidden="true" />{" "}
          <a href={`mailto:${user.item.email}`}>{user.item.email}</a>
        </p>
      </div>
    </>
  );
}

export default UserItem;
