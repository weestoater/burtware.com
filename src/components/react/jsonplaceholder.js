import React, { useState, useEffect } from 'react';

export const JsonPlaceholder = () => {

    // const fetchData = fetch('https://geodb-cities-graphql.p.rapidapi.com', {
    //     method: 'POST',
    //     headers: {'Content-Type': 'application/json'},
    //     body: JSON.stringify({
    //         query:`{
    //             countries {
    //                 name
    //                 capital
    //                 currency
    //             }
    //         }`
    //     })
    // })
    // .then(res => res.json())
    // .then(res => console.log(res.data));

    // fetchData();

    return (
        <div className="card">
            <div className="card-header">FETCH DATA</div>
            <div className="card-body">
                body    
            </div> 
        </div>
    );
}