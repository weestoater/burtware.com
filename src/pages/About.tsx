import React from "react";
import AboutContent from "../components/about/AboutContent";

export default function About() {
  return (
    <div className="row">
      <AboutContent />
    </div>
  );
}
