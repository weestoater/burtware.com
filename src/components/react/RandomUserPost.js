import React, {useState, useEffect} from 'react';
import getApi from '../../adapters/commonApi';
import Spinner from '../global/Spinner';
import UserItem from './UserItem';

export default function RandomUsersPost() {

    const [users, setUsers] = useState({});
    const apiUrl = "https://randomuser.me/api/?results=3";

    useEffect(() => {
        getApi(apiUrl).then(setUsers);
      }, []);

    return (
        <div className="col-lg-4 col-md-6 mb-4">
            <div className="card">
                <div className="card-header">
                    <i className="bi bi-filetype-jsx" />
                    <span className="pl-4">Random Users API</span>
                </div>
                <div className="card-body users">
                {users.results ? (
                    users.results.map((user, key) => <UserItem key={key} item={user} /> )
                ) : ( 
                <Spinner /> 
                )}
                </div>
            </div>
        </div>
    );
}
