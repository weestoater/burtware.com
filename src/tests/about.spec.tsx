import React from "react";
import { render, screen } from "@testing-library/react";
import About from "../pages/About";

describe('Testing About page', () => {

    it('can render About', () => {
        render(<About />);
        const burtware = screen.getByText(/about burtware/i);
        const cypress = screen.getByText(/cypress/i);
        expect(burtware).toBeInTheDocument();
        expect(cypress).toBeInTheDocument();
    });
});