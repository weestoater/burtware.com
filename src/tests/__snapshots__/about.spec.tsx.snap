// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing About page can render AboutPage 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-sm-12"
    >
      <h1
        class="visually-hidden"
      >
        About
      </h1>
    </div>
    <div
      class="col-lg-4 col-md-6"
    >
      <div
        class="card mb-4"
      >
        <div
          class="card-header"
        >
          <i
            class="bi bi-wrench-adjustable-circle"
          />
           About burtware
        </div>
        <div
          class="card-body"
        >
          <img
            alt=""
            class="float-end img-thumbnail shadow ms-2 mb-2"
            src="burt.png"
          />
          <p>
            This is a site for me to play with and learn more about 
            <code>
              ReactJS
            </code>
             and 
            <code>
              AWS
            </code>
            . I use both technologies in my day job, as well as Angular; Node; Cypress and multiple CSS frameworks.
          </p>
          <p>
            I also want to use it to trial different experiments and implement the fabulous React Testing Library based on the amazing 
            <a
              href="http://www.testingjavascript.com"
              rel="noreferrer"
              target="_blank"
            >
              testing javascript
            </a>
             course from 
            <b>
              Kent C. Dodds
            </b>
            .
          </p>
          <p>
            I have also been working very heavily in Accessibility and I would like to flex some testing muscles too.
          </p>
        </div>
      </div>
    </div>
    <div
      class="col-lg-4 col-md-6"
    >
      <div
        class="card mb-4"
      >
        <div
          class="card-header"
        >
          <i
            class="bi bi-balloon-heart"
          />
           burtware CV
        </div>
        <div
          class="card-body"
        >
          <div
            class="accordion"
            id="joblist"
          >
            <div
              class="accordion-item"
            >
              <h2
                class="accordion-header"
                id="heading0"
              >
                <button
                  aria-controls="collapse0"
                  aria-expanded="false"
                  class="accordion-button collapsed"
                  data-bs-target="#collapse0"
                  data-bs-toggle="collapse"
                  type="button"
                >
                  <small>
                    Sep '16
                     - 
                    Today
                  </small>
                   
                  Lead UI Developer
                   · 
                  J.P. Morgan Chase & Co.
                </button>
              </h2>
              <div
                aria-labelledby="heading0"
                class="accordion-collapse collapse"
                data-bs-parent="#joblist"
                id="collapse0"
              >
                <div
                  class="accordion-body"
                >
                  Lead UI developer in a team of UX designers and researchers.  Subject matter expert Front End development / Accessibility + Web standards. Advising development teams on improvements to their UI development standards + practices; Assisting teams with Accessibility improvements; Encouraging / enabling adoption of standardised UI frameworks; Working with designers to understand implemenation issues / best practices; Advising on best use of framework components; Evangelising on Accessibility and web standards.  I have worked in other teams within the firm as UI Senior Associate, mentoring junior talent and leading on interface development. As well as working on voluntary projects such as Code for Good; Force for Good and Generation Tech - all of which help support young talent discover paths into IT.
                </div>
              </div>
            </div>
            <div
              class="accordion-item"
            >
              <h2
                class="accordion-header"
                id="heading1"
              >
                <button
                  aria-controls="collapse1"
                  aria-expanded="false"
                  class="accordion-button collapsed"
                  data-bs-target="#collapse1"
                  data-bs-toggle="collapse"
                  type="button"
                >
                  <small>
                    Aug '16
                     - 
                    Sep '16
                  </small>
                   
                  UI Developer
                   · 
                  Student Loans Co.
                </button>
              </h2>
              <div
                aria-labelledby="heading1"
                class="accordion-collapse collapse"
                data-bs-parent="#joblist"
                id="collapse1"
              >
                <div
                  class="accordion-body"
                >
                  Web developer in small team producing standards compliant, highly accessible content updates.
                </div>
              </div>
            </div>
            <div
              class="accordion-item"
            >
              <h2
                class="accordion-header"
                id="heading2"
              >
                <button
                  aria-controls="collapse2"
                  aria-expanded="false"
                  class="accordion-button collapsed"
                  data-bs-target="#collapse2"
                  data-bs-toggle="collapse"
                  type="button"
                >
                  <small>
                    Nov '14
                     - 
                    Jul '16
                  </small>
                   
                  Senior Front-End Developer / UI Team Lead
                   · 
                  Xedo
                </button>
              </h2>
              <div
                aria-labelledby="heading2"
                class="accordion-collapse collapse"
                data-bs-parent="#joblist"
                id="collapse2"
              >
                <div
                  class="accordion-body"
                >
                  Implemented Umbraco to power content in the core app and upgraded legacy sites. Agile methodology, coached by the excellent Chris McDermott and learned the benefits of the empowered autonomous teams. Promoted to the UX /UI Team Lead role: I encouraged Jasmine TDD JavaScript development in Xedo and Visual Regression Testing for the UX/UI.
                </div>
              </div>
            </div>
            <div
              class="accordion-item"
            >
              <h2
                class="accordion-header"
                id="heading3"
              >
                <button
                  aria-controls="collapse3"
                  aria-expanded="false"
                  class="accordion-button collapsed"
                  data-bs-target="#collapse3"
                  data-bs-toggle="collapse"
                  type="button"
                >
                  <small>
                    Dec '11
                     - 
                    Nov '14
                  </small>
                   
                  Senior Web Developer / Designer
                   · 
                  IRW Systems Ltd.
                </button>
              </h2>
              <div
                aria-labelledby="heading3"
                class="accordion-collapse collapse"
                data-bs-parent="#joblist"
                id="collapse3"
              >
                <div
                  class="accordion-body"
                >
                  Front-End specialist within team of .NET developers. Worked on numerous interfaces for applications; certified Umbraco developer and produced many responsive websites using Umbraco;  Sharepoint interface development - wrestling with the nested CSS; Gained Microsoft Certifications - MCSD Web Applications and bing accredited
                </div>
              </div>
            </div>
            <div
              class="accordion-item"
            >
              <h2
                class="accordion-header"
                id="heading4"
              >
                <button
                  aria-controls="collapse4"
                  aria-expanded="false"
                  class="accordion-button collapsed"
                  data-bs-target="#collapse4"
                  data-bs-toggle="collapse"
                  type="button"
                >
                  <small>
                    Jan '08
                     - 
                    Dec '11
                  </small>
                   
                  Senior Web Developer
                   · 
                  East Renfrewshire Council
                </button>
              </h2>
              <div
                aria-labelledby="heading4"
                class="accordion-collapse collapse"
                data-bs-parent="#joblist"
                id="collapse4"
              >
                <div
                  class="accordion-body"
                >
                  Maintenance and update of Council website. Build and training of editors for new Council intranet and website, using .NET based GOSS CMS. Accessibility advocate.
                </div>
              </div>
            </div>
            <div
              class="accordion-item"
            >
              <h2
                class="accordion-header"
                id="heading5"
              >
                <button
                  aria-controls="collapse5"
                  aria-expanded="false"
                  class="accordion-button collapsed"
                  data-bs-target="#collapse5"
                  data-bs-toggle="collapse"
                  type="button"
                >
                  <small>
                    Aug '07
                     - 
                    Oct '07
                  </small>
                   
                  Freelance
                   · 
                  Education Scotland
                </button>
              </h2>
              <div
                aria-labelledby="heading5"
                class="accordion-collapse collapse"
                data-bs-parent="#joblist"
                id="collapse5"
              >
                <div
                  class="accordion-body"
                >
                  I worked on a SQL server migration project, requiring analysis and accuracy.
                </div>
              </div>
            </div>
            <div
              class="accordion-item"
            >
              <h2
                class="accordion-header"
                id="heading6"
              >
                <button
                  aria-controls="collapse6"
                  aria-expanded="false"
                  class="accordion-button collapsed"
                  data-bs-target="#collapse6"
                  data-bs-toggle="collapse"
                  type="button"
                >
                  <small>
                    Feb '05
                     - 
                    Sep '07
                  </small>
                   
                  Freelance
                   · 
                  accessibilitygnu.com
                </button>
              </h2>
              <div
                aria-labelledby="heading6"
                class="accordion-collapse collapse"
                data-bs-parent="#joblist"
                id="collapse6"
              >
                <div
                  class="accordion-body"
                >
                  Taking a career break from UK, worked out of South Korea doing TEfL classes during day and contracting on websites and design in my spare time.  Built up a client base of repeat business and some successful outcomes.
                </div>
              </div>
            </div>
            <div
              class="accordion-item"
            >
              <h2
                class="accordion-header"
                id="heading7"
              >
                <button
                  aria-controls="collapse7"
                  aria-expanded="false"
                  class="accordion-button collapsed"
                  data-bs-target="#collapse7"
                  data-bs-toggle="collapse"
                  type="button"
                >
                  <small>
                    Sep '00
                     - 
                    Feb '05
                  </small>
                   
                  Web Developer
                   · 
                  Learning & Teaching Scotland
                </button>
              </h2>
              <div
                aria-labelledby="heading7"
                class="accordion-collapse collapse"
                data-bs-parent="#joblist"
                id="collapse7"
              >
                <div
                  class="accordion-body"
                >
                  Worked on a variety of educational projects, from html driven information sites to highly interactive Flash game site which won National Award.  Also began journey of discovery into Accessibility and became expert in web standards promoting best practices across teams.
                </div>
              </div>
            </div>
            <div
              class="accordion-item"
            >
              <h2
                class="accordion-header"
                id="heading8"
              >
                <button
                  aria-controls="collapse8"
                  aria-expanded="false"
                  class="accordion-button collapsed"
                  data-bs-target="#collapse8"
                  data-bs-toggle="collapse"
                  type="button"
                >
                  <small>
                    Mar '96
                     - 
                    Sep '00
                  </small>
                   
                  DBA / Web Developer
                   · 
                  DCA Data Services Ltd.
                </button>
              </h2>
              <div
                aria-labelledby="heading8"
                class="accordion-collapse collapse"
                data-bs-parent="#joblist"
                id="collapse8"
              >
                <div
                  class="accordion-body"
                >
                  Small consultancy requiring me to wear many hats. Database Administrator for National Training Information DB; Web Designer and Developer; Network Administrator for office hardware.
                </div>
              </div>
            </div>
            <div
              class="accordion-item"
            >
              <h2
                class="accordion-header"
                id="heading9"
              >
                <button
                  aria-controls="collapse9"
                  aria-expanded="false"
                  class="accordion-button collapsed"
                  data-bs-target="#collapse9"
                  data-bs-toggle="collapse"
                  type="button"
                >
                  <small>
                    Jul '95
                     - 
                    Feb '96
                  </small>
                   
                  Graduate Developer
                   · 
                  Rockmoor Ltd.
                </button>
              </h2>
              <div
                aria-labelledby="heading9"
                class="accordion-collapse collapse"
                data-bs-parent="#joblist"
                id="collapse9"
              >
                <div
                  class="accordion-body"
                >
                  Worked on a number of interface designs for Eurodesk project; desktop publishing for the Scottish Music Information Centre handbook and icon designs for CiPFA application.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
